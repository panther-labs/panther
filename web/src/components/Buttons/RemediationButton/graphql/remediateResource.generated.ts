/* eslint-disable import/order, import/no-duplicates */
import * as Types from '../../../../../__generated__/schema';

import gql from 'graphql-tag';
import * as ApolloReactCommon from '@apollo/client';
import * as ApolloReactHooks from '@apollo/client';

export type RemediateResourceVariables = {
  input: Types.RemediateResourceInput;
};

export type RemediateResource = Pick<Types.Mutation, 'remediateResource'>;

export const RemediateResourceDocument = gql`
  mutation RemediateResource($input: RemediateResourceInput!) {
    remediateResource(input: $input)
  }
`;
export type RemediateResourceMutationFn = ApolloReactCommon.MutationFunction<
  RemediateResource,
  RemediateResourceVariables
>;

/**
 * __useRemediateResource__
 *
 * To run a mutation, you first call `useRemediateResource` within a React component and pass it any options that fit your needs.
 * When your component renders, `useRemediateResource` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [remediateResource, { data, loading, error }] = useRemediateResource({
 *   variables: {
 *      input: // value for 'input'
 *   },
 * });
 */
export function useRemediateResource(
  baseOptions?: ApolloReactHooks.MutationHookOptions<RemediateResource, RemediateResourceVariables>
) {
  return ApolloReactHooks.useMutation<RemediateResource, RemediateResourceVariables>(
    RemediateResourceDocument,
    baseOptions
  );
}
export type RemediateResourceHookResult = ReturnType<typeof useRemediateResource>;
export type RemediateResourceMutationResult = ApolloReactCommon.MutationResult<RemediateResource>;
export type RemediateResourceMutationOptions = ApolloReactCommon.BaseMutationOptions<
  RemediateResource,
  RemediateResourceVariables
>;
