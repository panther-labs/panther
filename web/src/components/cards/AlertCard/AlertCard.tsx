/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import GenericItemCard from 'Components/GenericItemCard';
import { Flex, Icon, Link, Text, Box } from 'pouncejs';
import { AlertTypesEnum } from 'Generated/schema';
import { Link as RRLink } from 'react-router-dom';
import SeverityBadge from 'Components/badges/SeverityBadge';
import React from 'react';
import urls from 'Source/urls';
import LinkButton from 'Components/buttons/LinkButton';
import RelatedDestinations from 'Components/RelatedDestinations';
import BulletedLogTypeList from 'Components/BulletedLogTypeList';
import { AlertSummaryFull } from 'Source/graphql/fragments/AlertSummaryFull.generated';
import { formatDatetime } from 'Helpers/utils';
import useAlertDestinations from 'Hooks/useAlertDestinations';
import useAlertDestinationsDeliverySuccess from 'Hooks/useAlertDestinationsDeliverySuccess';
import { SelectCheckbox } from 'Components/utils/SelectContext';
import UpdateAlertDropdown from '../../dropdowns/UpdateAlertDropdown';

interface AlertCardProps {
  alert: AlertSummaryFull;
  hideRuleButton?: boolean;
  selectionEnabled?: boolean;
}

const AlertCard: React.FC<AlertCardProps> = ({
  alert,
  hideRuleButton = false,
  selectionEnabled = false,
}) => {
  const { alertDestinations, loading: loadingDestinations } = useAlertDestinations({ alert });
  const { allDestinationDeliveredSuccessfully, loading } = useAlertDestinationsDeliverySuccess({
    alert,
  });

  return (
    <GenericItemCard borderColor={alert.type === AlertTypesEnum.RuleError ? 'red-600' : 'teal-400'}>
      <Flex align="start" pr={2}>
        {selectionEnabled && (
          <Box transform="translate3d(0,-8px,0)">
            <SelectCheckbox selectionId={alert.alertId} />
          </Box>
        )}
      </Flex>
      <GenericItemCard.Body>
        <GenericItemCard.Heading>
          <Link
            as={RRLink}
            aria-label="Link to Alert"
            to={urls.logAnalysis.alerts.details(alert.alertId)}
          >
            {alert.title}
          </Link>
        </GenericItemCard.Heading>
        <GenericItemCard.ValuesGroup>
          {!hideRuleButton && (
            <GenericItemCard.Value
              value={
                <LinkButton
                  aria-label="Link to Rule"
                  to={urls.logAnalysis.rules.details(alert.ruleId)}
                  variantColor="navyblue"
                  size="medium"
                >
                  View Rule
                </LinkButton>
              }
            />
          )}
          <GenericItemCard.Value
            label="Alert Type"
            value={
              <Text color={alert.type === AlertTypesEnum.RuleError ? 'red-500' : 'teal-100'}>
                {alert.type === AlertTypesEnum.RuleError ? 'Rule Error' : 'Rule Match'}
              </Text>
            }
          />
          <GenericItemCard.Value
            label="Destinations"
            value={
              <RelatedDestinations destinations={alertDestinations} loading={loadingDestinations} />
            }
          />
          <GenericItemCard.Value
            label="Log Types"
            value={<BulletedLogTypeList logTypes={alert.logTypes} limit={2} />}
          />
          <GenericItemCard.Value
            label="Events"
            value={alert?.eventsMatched ? alert?.eventsMatched.toLocaleString() : '0'}
          />
          <GenericItemCard.Value label="Time Created" value={formatDatetime(alert.creationTime)} />
          <Flex ml="auto" mr={0} align="flex-end" spacing={2}>
            <SeverityBadge severity={alert.severity} />
            <UpdateAlertDropdown alert={alert} />
          </Flex>
        </GenericItemCard.ValuesGroup>
        {!loading && !allDestinationDeliveredSuccessfully && (
          <Flex
            as="section"
            align="center"
            spacing={2}
            mt={2}
            aria-label="Destination delivery failure"
          >
            <Icon type="alert-circle-filled" size="medium" color="red-300" />
            <Text fontSize="small" fontStyle="italic" color="red-300">
              There was an issue with the delivery of this alert to a selected destination. See
              specific Alert for details.
            </Text>
          </Flex>
        )}
      </GenericItemCard.Body>
    </GenericItemCard>
  );
};

export default React.memo(AlertCard);
