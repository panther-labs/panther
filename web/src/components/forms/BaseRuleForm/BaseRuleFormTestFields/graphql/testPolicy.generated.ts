/* eslint-disable import/order, import/no-duplicates */
import * as Types from '../../../../../../__generated__/schema';

import gql from 'graphql-tag';
import * as ApolloReactCommon from '@apollo/client';
import * as ApolloReactHooks from '@apollo/client';

export type TestPolicyVariables = {
  input?: Types.Maybe<Types.TestPolicyInput>;
};

export type TestPolicy = {
  testPolicy: Types.Maybe<
    Pick<Types.TestPolicyResponse, 'testSummary' | 'testsPassed' | 'testsFailed'> & {
      testsErrored: Types.Maybe<
        Array<Types.Maybe<Pick<Types.PolicyUnitTestError, 'errorMessage' | 'name'>>>
      >;
    }
  >;
};

export const TestPolicyDocument = gql`
  mutation TestPolicy($input: TestPolicyInput) {
    testPolicy(input: $input) {
      testSummary
      testsPassed
      testsFailed
      testsErrored {
        errorMessage
        name
      }
    }
  }
`;
export type TestPolicyMutationFn = ApolloReactCommon.MutationFunction<
  TestPolicy,
  TestPolicyVariables
>;

/**
 * __useTestPolicy__
 *
 * To run a mutation, you first call `useTestPolicy` within a React component and pass it any options that fit your needs.
 * When your component renders, `useTestPolicy` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [testPolicy, { data, loading, error }] = useTestPolicy({
 *   variables: {
 *      input: // value for 'input'
 *   },
 * });
 */
export function useTestPolicy(
  baseOptions?: ApolloReactHooks.MutationHookOptions<TestPolicy, TestPolicyVariables>
) {
  return ApolloReactHooks.useMutation<TestPolicy, TestPolicyVariables>(
    TestPolicyDocument,
    baseOptions
  );
}
export type TestPolicyHookResult = ReturnType<typeof useTestPolicy>;
export type TestPolicyMutationResult = ApolloReactCommon.MutationResult<TestPolicy>;
export type TestPolicyMutationOptions = ApolloReactCommon.BaseMutationOptions<
  TestPolicy,
  TestPolicyVariables
>;
