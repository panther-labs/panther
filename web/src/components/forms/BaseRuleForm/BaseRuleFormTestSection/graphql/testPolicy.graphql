mutation TestPolicy($input: TestPolicyInput) {
    testPolicy(input: $input) {
        testSummary
        testsPassed
        testsFailed
        testsErrored {
            errorMessage
            name
        }
    }
}