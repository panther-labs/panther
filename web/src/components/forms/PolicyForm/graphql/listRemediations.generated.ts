/* eslint-disable import/order, import/no-duplicates */
import * as Types from '../../../../../__generated__/schema';

import gql from 'graphql-tag';
import * as ApolloReactCommon from '@apollo/client';
import * as ApolloReactHooks from '@apollo/client';

export type ListRemediationsVariables = {};

export type ListRemediations = Pick<Types.Query, 'remediations'>;

export const ListRemediationsDocument = gql`
  query ListRemediations {
    remediations
  }
`;

/**
 * __useListRemediations__
 *
 * To run a query within a React component, call `useListRemediations` and pass it any options that fit your needs.
 * When your component renders, `useListRemediations` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useListRemediations({
 *   variables: {
 *   },
 * });
 */
export function useListRemediations(
  baseOptions?: ApolloReactHooks.QueryHookOptions<ListRemediations, ListRemediationsVariables>
) {
  return ApolloReactHooks.useQuery<ListRemediations, ListRemediationsVariables>(
    ListRemediationsDocument,
    baseOptions
  );
}
export function useListRemediationsLazyQuery(
  baseOptions?: ApolloReactHooks.LazyQueryHookOptions<ListRemediations, ListRemediationsVariables>
) {
  return ApolloReactHooks.useLazyQuery<ListRemediations, ListRemediationsVariables>(
    ListRemediationsDocument,
    baseOptions
  );
}
export type ListRemediationsHookResult = ReturnType<typeof useListRemediations>;
export type ListRemediationsLazyQueryHookResult = ReturnType<typeof useListRemediationsLazyQuery>;
export type ListRemediationsQueryResult = ApolloReactCommon.QueryResult<
  ListRemediations,
  ListRemediationsVariables
>;
