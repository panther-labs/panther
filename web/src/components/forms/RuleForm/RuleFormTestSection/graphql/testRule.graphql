mutation TestRule($input: TestRuleInput!) {
  testRule(input: $input) {
    results {
      id
      name
      passed
      error {
        message
      }
      functions {
        ruleFunction {
          output
          error {
            message
          }
        }
        titleFunction {
          output
          error {
            message
          }
        }
        dedupFunction {
          output
          error {
            message
          }
        }
      }
    }
  }
}
