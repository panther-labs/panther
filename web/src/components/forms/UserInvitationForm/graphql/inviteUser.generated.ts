/* eslint-disable import/order, import/no-duplicates */
import * as Types from '../../../../../__generated__/schema';

import gql from 'graphql-tag';
import * as ApolloReactCommon from '@apollo/client';
import * as ApolloReactHooks from '@apollo/client';

export type InviteUserVariables = {
  input: Types.InviteUserInput;
};

export type InviteUser = { inviteUser: Types.Maybe<Pick<Types.InviteUserResponse, 'id'>> };

export const InviteUserDocument = gql`
  mutation InviteUser($input: InviteUserInput!) {
    inviteUser(input: $input) {
      id
    }
  }
`;
export type InviteUserMutationFn = ApolloReactCommon.MutationFunction<
  InviteUser,
  InviteUserVariables
>;

/**
 * __useInviteUser__
 *
 * To run a mutation, you first call `useInviteUser` within a React component and pass it any options that fit your needs.
 * When your component renders, `useInviteUser` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [inviteUser, { data, loading, error }] = useInviteUser({
 *   variables: {
 *      input: // value for 'input'
 *   },
 * });
 */
export function useInviteUser(
  baseOptions?: ApolloReactHooks.MutationHookOptions<InviteUser, InviteUserVariables>
) {
  return ApolloReactHooks.useMutation<InviteUser, InviteUserVariables>(
    InviteUserDocument,
    baseOptions
  );
}
export type InviteUserHookResult = ReturnType<typeof useInviteUser>;
export type InviteUserMutationResult = ApolloReactCommon.MutationResult<InviteUser>;
export type InviteUserMutationOptions = ApolloReactCommon.BaseMutationOptions<
  InviteUser,
  InviteUserVariables
>;
