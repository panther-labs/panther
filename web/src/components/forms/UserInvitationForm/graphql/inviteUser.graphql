mutation InviteUser($input: InviteUserInput!) {
    inviteUser(input: $input) {
        id
    }
}