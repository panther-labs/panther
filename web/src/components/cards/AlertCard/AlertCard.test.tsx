/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import {
  buildAlertSummary,
  buildDeliveryResponse,
  buildDestination,
  render,
  waitForElementToBeRemoved,
} from 'test-utils';
import React from 'react';
import { AlertStatusesEnum, DestinationTypeEnum, SeverityEnum } from 'Generated/schema';
import urls from 'Source/urls';
import { mockListDestinations } from 'Source/graphql/queries';
import AlertCard from './index';

describe('AlertCard', () => {
  it('should match snapshot', async () => {
    const alertData = buildAlertSummary();

    const { container } = render(<AlertCard alert={alertData} />);

    expect(container).toMatchSnapshot();
  });

  it('displays the correct Alert data in the card', async () => {
    const alertData = buildAlertSummary();

    const { getByText, getByAriaLabel } = render(<AlertCard alert={alertData} />);

    expect(getByText(alertData.title)).toBeInTheDocument();
    expect(getByAriaLabel(`Link to rule ${alertData.ruleId}`)).toBeInTheDocument();
    expect(getByText('Events')).toBeInTheDocument();
    expect(getByText('Destinations')).toBeInTheDocument();
    expect(getByAriaLabel(`Creation time for ${alertData.alertId}`)).toBeInTheDocument();
    expect(getByText(SeverityEnum.Medium)).toBeInTheDocument();
    expect(getByText(AlertStatusesEnum.Triaged)).toBeInTheDocument();
    expect(getByAriaLabel('Change Alert Status')).toBeInTheDocument();
  });

  it('should not display link to Rule', async () => {
    const alertData = buildAlertSummary();

    const { queryByText, queryByAriaLabel } = render(
      <AlertCard alert={alertData} hideRuleButton />
    );

    expect(queryByText(alertData.title)).toBeInTheDocument();
    expect(queryByAriaLabel(`Link to rule ${alertData.ruleId}`)).not.toBeInTheDocument();
  });

  it('should check links are valid', async () => {
    const alertData = buildAlertSummary();
    const { getByAriaLabel } = render(<AlertCard alert={alertData} />);
    expect(getByAriaLabel('Link to Alert')).toHaveAttribute(
      'href',
      urls.logAnalysis.alerts.details(alertData.alertId)
    );
    expect(getByAriaLabel(`Link to rule ${alertData.ruleId}`)).toHaveAttribute(
      'href',
      urls.logAnalysis.rules.details(alertData.ruleId)
    );
  });

  it('should render alert destinations logos', async () => {
    const outputId = 'destination-of-alert';
    const alertData = buildAlertSummary({
      deliveryResponses: [buildDeliveryResponse({ outputId })],
    });
    const destination = buildDestination({ outputId, outputType: DestinationTypeEnum.Slack });
    const mocks = [mockListDestinations({ data: { destinations: [destination] } })];

    const { getByAriaLabel, getByAltText } = render(<AlertCard alert={alertData} />, {
      mocks,
    });
    const loadingInterfaceElement = getByAriaLabel('Loading...');
    expect(loadingInterfaceElement).toBeInTheDocument();
    await waitForElementToBeRemoved(loadingInterfaceElement);
    expect(getByAltText(`${destination.outputType} logo`)).toBeInTheDocument();
  });

  it('should render message that destination delivery is failing', async () => {
    const outputId = 'destination-of-alert';
    const alertData = buildAlertSummary({
      deliveryResponses: [buildDeliveryResponse({ outputId, success: false })],
    });
    const destination = buildDestination({ outputId, outputType: DestinationTypeEnum.Slack });
    const mocks = [mockListDestinations({ data: { destinations: [destination] } })];

    const { getByAriaLabel, getByAltText } = render(<AlertCard alert={alertData} />, {
      mocks,
    });
    const loadingInterfaceElement = getByAriaLabel('Loading...');
    expect(loadingInterfaceElement).toBeInTheDocument();
    await waitForElementToBeRemoved(loadingInterfaceElement);
    expect(getByAltText(`${destination.outputType} logo`)).toBeInTheDocument();
    expect(getByAriaLabel('Destination delivery failure')).toBeInTheDocument();
  });
});
