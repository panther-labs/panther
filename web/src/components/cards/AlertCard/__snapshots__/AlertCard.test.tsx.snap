// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertCard should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0);
    -moz-transform: rotate(0);
    -ms-transform: rotate(0);
    transform: rotate(0);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-1 {
  0% {
    stroke-dashoffset: 600;
  }

  100% {
    stroke-dashoffset: 0;
  }
}

.panther-41 {
  background-color: #1d2a3f;
  border-radius: 4px;
  padding: 16px;
  overflow: hidden;
}



.panther-39 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  height: 100%;
}

.panther-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  padding-right: 8px;
}

.panther-38 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
}

.panther-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin-bottom: 8px;
}

.panther-2 {
  font-weight: 500;
  margin-right: auto;
  max-width: 70%;
  word-break: break-word;
}

.panther-1 {
  -webkit-transition: color 0.1s ease-out;
  transition: color 0.1s ease-out;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #7fd3ff;
}

.panther-1:hover {
  color: #ace2ff;
}

.panther-1:focus {
  color: #ace2ff;
}

.panther-1:active,
.panther-1[data-active=true] {
  color: #11a5f3;
}

.panther-3 {
  font-size: 0.75rem;
  color: #716f6f;
}

.panther-5 {
  font-size: 0.75rem;
  color: #d64242;
}

.panther-37 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.panther-37>*:not(:last-child) {
  margin-right: 32px;
}

.panther-12 {
  margin-top: 8px;
  margin-bottom: 8px;
}

.panther-6 {
  color: #bdbdbd;
  font-size: 0.563rem;
  font-weight: 500;
}

.panther-11 {
  font-size: 0.875rem;
  font-weight: 500;
  opacity: 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 24px;
}

.panther-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.panther-10>*:not(:last-child) {
  margin-right: 8px;
}

.panther-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.panther-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 24px;
  height: 24px;
  background-color: #4f627c;
  border-color: #4f627c;
  border-radius: 99999px;
}

.panther-9:hover {
  background-color: #11a5f3;
  border-color: #11a5f3;
}

.panther-8 {
  display: inline-block;
  vertical-align: sub;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 12px;
  height: 12px;
  color: currentColor;
}

.panther-16 {
  height: 18px;
}

.panther-15 {
  display: inline-block;
  vertical-align: sub;
  -webkit-animation: animation-0 2s linear infinite;
  animation: animation-0 2s linear infinite;
  width: 18px;
  height: 18px;
}

.panther-14 {
  stroke-dashoffset: 600;
  stroke-dasharray: 300;
  stroke-width: 13;
  stroke-miterlimit: 10;
  stroke-linecap: round;
  -webkit-animation: animation-1 1.6s cubic-bezier(0.4, 0.15, 0.6, 0.85) infinite;
  animation: animation-1 1.6s cubic-bezier(0.4, 0.15, 0.6, 0.85) infinite;
  stroke: currentColor;
  fill: transparent;
}

.panther-23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.panther-23>*:not(:last-child) {
  margin-right: 8px;
}

.panther-22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.panther-22>*:not(:last-child) {
  margin-right: 8px;
}

.panther-20 {
  width: 12px;
  height: 12px;
  background-color: hsl(353, 43%, 63%);
  border-radius: 99999px;
}

.panther-21 {
  font-size: 0.75rem;
  font-weight: 400;
  line-height: 1.5;
}

.panther-36 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  margin-left: auto;
  margin-right: 0;
}

.panther-36>*:not(:last-child) {
  margin-right: 8px;
}

.panther-29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: default;
  width: -webkit-fit-content;
  width: -moz-content;
  width: fit-content;
  min-width: 85px;
  text-align: center;
  font-weight: 700;
  border-radius: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-size: 0.75rem;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  border: 1px solid;
  border-color: #ffa500;
  background-color: transparent;
}

.panther-35 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.panther-35>*:not(:last-child) {
  margin-right: 4px;
}

.panther-31 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 85px;
}

.panther-30 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: default;
  width: -webkit-fit-content;
  width: -moz-content;
  width: fit-content;
  min-width: 85px;
  text-align: center;
  font-weight: 700;
  border-radius: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-size: 0.75rem;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  background-color: #6e3078;
}

.panther-34 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.panther-33 {
  outline: none;
  line-height: 1;
  height: 24px;
  width: 24px;
  -webkit-transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border-radius: 99999px;
  border: 1px solid;
  border-color: #3e516b;
  background-color: transparent;
  cursor: pointer;
  color: #f6f6f6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.panther-33:hover {
  background-color: #465973;
  border-color: #465973;
}

.panther-33:focus {
  background-color: #465973;
  border-color: #93a6c0;
}

.panther-33:active,
.panther-33[data-active=true] {
  background-color: #253852;
  border-color: #253852;
}

.panther-32 {
  display: inline-block;
  vertical-align: sub;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 16px;
  height: 16px;
  color: currentColor;
}

<div>
  <header
    id="main-header"
  />
  <section
    class="panther-41"
  >
    <div
      class="panther-40"
    >
      <div
        class="panther-39"
      >
        <div
          class="panther-0"
        />
        <div
          class="panther-38"
        >
          <header
            class="panther-4"
          >
            <h4
              class="panther-2"
            >
              <a
                aria-label="Link to Alert"
                class="panther-1"
                href="/log-analysis/alerts/f67b8f04-5fac-404a-93a4-38db29f258ba/"
              >
                indexing
              </a>
            </h4>
            <span
              aria-label="Creation time for f67b8f04-5fac-404a-93a4-38db29f258ba"
              class="panther-3"
            >
              2020-08-08 12:15 GMT
            </span>
          </header>
          <span
            class="panther-5"
          >
            Rule Match
          </span>
          <div
            class="panther-37"
          >
            <dl
              class="panther-12"
            >
              <dt
                aria-labelledby="ruleobject-object"
                class="panther-6"
              >
                Rule
              </dt>
              <dd
                aria-labelledby="ruleobject-object"
                class="panther-11"
              >
                <div
                  class="panther-10"
                >
                  <span
                    class="panther-7"
                  >
                    6eb9c948-5a13-4955-bd91-b98801b55bed
                  </span>
                  <a
                    aria-label="Link to rule 6eb9c948-5a13-4955-bd91-b98801b55bed"
                    href="/log-analysis/rules/6eb9c948-5a13-4955-bd91-b98801b55bed/"
                  >
                    <div
                      class="panther-9"
                    >
                      <svg
                        class="panther-8"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 4L10.59 5.41L16.17 11H4V13H16.17L10.59 18.59L12 20L20 12L12 4Z"
                        />
                      </svg>
                    </div>
                  </a>
                </div>
              </dd>
            </dl>
            <dl
              class="panther-12"
            >
              <dt
                aria-labelledby="destinationsobject-object"
                class="panther-6"
              >
                Destinations
              </dt>
              <dd
                aria-labelledby="destinationsobject-object"
                class="panther-11"
              >
                <div
                  class="panther-16"
                >
                  <svg
                    aria-label="Loading..."
                    class="panther-15"
                    viewBox="0 0 150 150"
                  >
                    <circle
                      class="panther-14"
                      color="current"
                      cx="75"
                      cy="75"
                      r="60"
                    />
                  </svg>
                </div>
              </dd>
            </dl>
            <dl
              class="panther-12"
            >
              <dt
                aria-labelledby="log-typesobject-object"
                class="panther-6"
              >
                Log Types
              </dt>
              <dd
                aria-labelledby="log-typesobject-object"
                class="panther-11"
              >
                <div
                  class="panther-23"
                >
                  <div
                    class="panther-22"
                  >
                    <span
                      class="panther-20"
                    />
                    <span
                      class="panther-21"
                    >
                      Costa Rica
                    </span>
                  </div>
                </div>
              </dd>
            </dl>
            <dl
              class="panther-12"
            >
              <dt
                aria-labelledby="events670"
                class="panther-6"
              >
                Events
              </dt>
              <dd
                aria-labelledby="events670"
                class="panther-11"
              >
                670
              </dd>
            </dl>
            <div
              class="panther-36"
            >
              <div
                aria-atomic="true"
                class="panther-29"
                role="status"
              >
                MEDIUM
              </div>
              <div
                class="panther-35"
              >
                <div
                  class="panther-31"
                >
                  <div
                    aria-atomic="true"
                    class="panther-30"
                    role="status"
                  >
                    TRIAGED
                  </div>
                </div>
                <div
                  aria-controls="menu--1"
                  aria-haspopup="true"
                  class="panther-34"
                  data-reach-menu-button=""
                  id="menu-button--menu--1"
                  type="button"
                >
                  <button
                    aria-label="Change Alert Status"
                    aria-pressed="false"
                    class="panther-33"
                    type="button"
                  >
                    <svg
                      class="panther-32"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10L12 15L17 10H7Z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <footer
    id="footer"
  />
</div>
`;
